### ISSUE TITLE
ports are open to localhost only

### ISSUE THREAD

**Issue by gorgia** [phusion/baseimage-docker#544]
I have noticed that exposed ports are reachable only from localhost.
In spite of the fact the the hosting systems itself shows that the container is listening againts 0.0.0.0.
```
FROM phusion/baseimage:latest
MAINTAINER Andrew Spear <gorgia@fastwebnet.it>

# Evironment variables
ENV DEBIAN_FRONTEND=noninteractive \
    OPENVPN_USERNAME=**None** \
    OPENVPN_PASSWORD=**None** \
    OPENVPN_PROVIDER=**None**

# Use baseimage-docker's init system.
CMD ["/sbin/my_init"]

# Update packages and install software
RUN apt-get update \
    && apt-get install -y openvpn inetutils-traceroute inetutils-ping wget curl \
    && curl -L https://github.com/jwilder/dockerize/releases/download/v0.2.0/dockerize-linux-amd64-v0.2.0.tar.gz | tar -C /usr/local/bin -xzv \
    && rm -rfv dockerize-linux-amd64-v0.2.0.tar.gz \
    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*


# Enabling SSH
RUN rm -f /etc/service/sshd/down

# Enabling the insecure key permanently
RUN /usr/sbin/enable_insecure_key

# Expose port and run
EXPOSE 22

# Volumes
VOLUME /config

# Adding utils scripts to bin
ADD bin/ /usr/local/bin/
RUN chmod +x /usr/local/bin/*

# Add configuration and scripts
ADD openvpn /etc/openvpn
RUN chmod +x /etc/openvpn/bin/* \
    && mkdir -p /etc/openvpn/up \
    && mkdir -p /etc/openvpn/down \
    && ln -s /usr/local/bin/ssh-restart /etc/openvpn/up/00-ssh-restart \
    && ln -s /usr/local/bin/my-public-ip-info /etc/openvpn/up/01-my-public-ip-info

# Running scripts during container startup
RUN mkdir -p /etc/my_init.d \
    && ln -s /etc/openvpn/bin/openvpn-setup.sh /etc/my_init.d/openvpn-setup.sh \
    && chmod +x /etc/my_init.d/*

# Add to runit
RUN mkdir /etc/service/openvpn \
    && ln -s /etc/openvpn/bin/openvpn-run.sh /etc/service/openvpn/run \
    && ln -s /etc/openvpn/bin/openvpn-finish.sh /etc/service/openvpn/finish \
    && chmod +x /etc/service/openvpn/run \
    && chmod +x /etc/service/openvpn/finish
```
in the previous dockerfile if I use phusion/baseimage:master, the port 22 is reachable only from localhost.

**Comment by kingdonb** [comment]
Interesting. Are you able to bisect and narrow this down to a specific change?

From what I can tell, the "latest" tag was last updated 2 years ago, but you didn't mention that one.  The "master" tag is a bit ahead of 0.11, having been updated 5 months ago, when 0.11 was released about a year ago according to Docker Hub.

Can you test this with some more tag versions?  I'm skeptical that it's anything other than a Docker daemon configuration issue, but if you can demonstrate that something changed and maybe it was unintentional, then we will have something actionable to make a PR out of.

I'm not sure how to reproduce this, or what has changed, and would appreciate a bit more information if you think there is an issue.

**Comment by gorgia** [comment]
Hi,
It affect 0.11 and master as well, on fresh docker install.

**Comment by Theaxiom** [comment]
@kingdonb maybe `latest` tag should reflect the last released version.

**Comment by sshishov** [comment]
Hello @gorgia , could you please elaborate how you are testing if the port is open or not?
Make sure that you are testing from the container in the same network (in docker).
If you connect from local computer, make sure to forward port from local to container.

Looking forward to help you here.

**Comment by gorgia** [comment]
Hello @sshishov,
What I can say to you, after almost two year, is that I was unable to make it work. There was really no reason why the port accept connection only from localhost. I tested them with basic nmap.
So I sadly switched to ubuntu base image and with the same config everything has been working ever since.

**Comment by sshishov** [comment]
@gorgia Thanks for your input. I tried to replicate the issue but for me I could connect to the container on port 22 from another container running inside same docker network. I have tested on `master` and on `focal-1.0.0`.
If you are not facing this issue now and we cannot reproduce it, would it be possible for you to close the issue with appropriate comment. In this case if someone else will face the same issue, he will just reopen it. Sounds fair?
